docs: clarify usage stats env vars for Core 1.11+ and 2.x (#9658)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/usage-stats.md

- An anonymized hash key representing the raw model content.
- Number of nodes that were run.

<VersionBlock lastVersion="1.99">

## dbt Core telemetry

<Constant name="core" /> has telemetry enabled by default. For full transparency, you can see all the event definitions in [`tracking.py`](https://github.com/dbt-labs/dbt-core/blob/1.latest/core/dbt/tracking.py).

You can opt out of event tracking at any time by adding the following configuration to your `dbt_project.yml` file:

<File name="dbt_project.yml">

```yaml
flags:
  send_anonymous_usage_stats: false
```

</File>

> You can also set the `DO_NOT_TRACK` environment variable. For more information, refer to [Environment variables](/docs/build/environment-variables).

Reviewer: @QMalcolm is this one not prefixed with DBT_ENGINE_ for some reason?

Reviewer 2: Woof. I went and tracked this down. `DO_NOT_TRACK` is handled differently than every other environment variable we have in the python codebase. It looks like it got missed when we made the shift to `DBT_ENGINE_` prefixes. This probably needs to be fixed in 2.x and 1.x versions 🫠

  dbt Core users can also use the `DO_NOT_TRACK` environment variable to enable or disable sending anonymous data. For more information, see [Environment variables](/docs/build/environment-variables).

>   `DO_NOT_TRACK=1` is the same as <VersionBlock lastVersion="1.10">`DBT_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock><VersionBlock firstVersion="1.11">`DBT_SEND_ANONYMOUS_USAGE_STATS=False` or `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock>.

Reviewer 3: Using the prefix is the correct option (and should be the one that’s documented):
```suggestion
`DO_NOT_TRACK=1` is the same as <VersionBlock lastVersion="1.10">`DBT_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock><VersionBlock firstVersion="1.11">`DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock>.
```

>   `DO_NOT_TRACK=1` is the same as <VersionBlock lastVersion="1.10">`DBT_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock><VersionBlock firstVersion="1.11">`DBT_SEND_ANONYMOUS_USAGE_STATS=False` or `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock>.

Reviewer: We should just document the env vars with `DBT_ENGINE_` prefix, the legacy one is just for backwards compat

>   `DO_NOT_TRACK=1` is the same as <VersionBlock lastVersion="1.10">`DBT_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock><VersionBlock firstVersion="1.11">`DBT_SEND_ANONYMOUS_USAGE_STATS=False` or `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock>.

Reviewer: ```suggestion
  `DO_NOT_TRACK=1` is the same as <VersionBlock lastVersion="1.10">`DBT_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock><VersionBlock firstVersion="1.11"> `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS=False`</VersionBlock>.
```

  <VersionBlock firstVersion="1.11">

>   In dbt Core 1.11 and later (including dbt Core 2.x), both `DBT_SEND_ANONYMOUS_USAGE_STATS` and `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS` are accepted environment variables for the same setting. `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS` follows the engine-scoped naming convention used in the [global configs reference](/reference/global-configs/about-global-configs).

Reviewer 3: ```suggestion
Starting in <Constant name="core" /> 1.11, both `DBT_SEND_ANONYMOUS_USAGE_STATS` and `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS` are accepted. `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS` is the preferred variable and takes precedence when both are set.
```

Reviewer 3: @graciegoheen what do you think about adding this?

>   In dbt Core 1.11 and later (including dbt Core 2.x), both `DBT_SEND_ANONYMOUS_USAGE_STATS` and `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS` are accepted environment variables for the same setting. `DBT_ENGINE_SEND_ANONYMOUS_USAGE_STATS` follows the engine-scoped naming convention used in the [global configs reference](/reference/global-configs/about-global-configs).

Reviewer: ```suggestion
```

I don't think we need this callout
